lib/guardian/plug/verify_header.ex

```diff
@@ -1,90 +1,121 @@
+    It requires an implementation module, an error handler and a key.
 
-      plug Guardian.Plug.VerifyHeader, key: :secret
+    These can be set either:
 
-  Verifying the session will update the claims on the request,
-  available with Guardian.Plug.claims/1
+    1. Upstream on the connection with `plug Guardian.Pipeline`
+    2. Upstream on the connection with `Guardian.Pipeline.{put_module, put_error_handler, put_key}`
+    3. Inline with an option of `:module`, `:erorr_handler`, `:key`
 
-  In the case of an error, the claims will be set to { :error, reason }
+    If a token is found but is invalid, the error handler will be called with
+    `auth_error(conn, {:invalid_token, reason}, opts)`
 
-  A "realm" can be specified when using the plug.
-  Realms are like the name of the token and allow many tokens
-  to be sent with a single request.
+    Once a token has been found it will be decoded, the token and claims will be put onto the connection.
 
-      plug Guardian.Plug.VerifyHeader, realm: "Bearer"
+    They will be available using `Guardian.Plug.current_claims/2` and `Guardian.Plug.current_token/2`
 
-  When a realm is not specified,
-  the first authorization header found is used, and assumed to be a raw token
+    Options:
 
-  #### example
+    * `claims` - The literal claims to check to ensure that a token is valid
+    * `realm` - The prefix for the token in the Authorization header. Defaults to `bearer`. `:none` will not use a prefix.
+    * `key` - The location to store the information in the connection. Defaults to: `deafult`
 
-      plug Guardian.Plug.VerifyHeader
+    ### Example
 
-      # will take the first auth header
-      # Authorization: <jwt>
-  """
+    ```elixir
 
-  def init(opts \\ %{}) do
-    opts_map = Enum.into(opts, %{})
-    realm = Map.get(opts_map, :realm)
-    if realm do
-      {:ok, reg} = Regex.compile("#{realm}\:?\s+(.*)$", "i")
-      Map.put(opts_map, :realm_reg, reg)
-    else
-      opts_map
-    end
-  end
+    # setup the upstream pipeline
+
+    plug Guardian.Plug.VerifyHeader, claims: %{typ: "access"}
+
+    ```
+
+    This will check the authorization header for a token
+
+    `Authorization Bearer: <token>`
+
+    This token will be placed into the connection depending on the key and can be accessed with
+    `Guardian.Plug.current_token` and `Guardian.Plug.current_claims`
+
+    OR
+
+    `MyApp.ImplementationModule.current_token` and `MyApp.ImplementationModule.current_claims`
+    """
 
-  def call(conn, opts) do
-    key = Map.get(opts, :key, :default)
+    alias Guardian.Plug, as: GPlug
+    alias GPlug.Pipeline
 
-    case Guardian.Plug.claims(conn, key) do
-      {:ok, _} -> conn
-      {:error, :no_session} ->
-        verify_token(conn, fetch_token(conn, opts), key)
-      _ -> conn
+    import Plug.Conn
+
+    @spec init(Keyword.t) :: Keyword.t
+    def init(opts \\ []) do
+      realm = Keyword.get(opts, :realm, "bearer")
```

Comment:
I think it should be capitalized. The header keys are lowercased but the values are not. So this could be funky if we leave it lowercased.